src/query/stream/streams.cpp

```diff
@@ -490,13 +490,17 @@ Streams::StreamsMap::iterator Streams::CreateConsumer(StreamsMap &map, const std
                             retry_interval = interpreter_context_->config.stream_transaction_retry_interval](
                                const std::vector<typename TStream::Message> &messages) mutable {
     auto accessor = interpreter_context->db->Access();
+    // register new interpreter into interpreter_context_
+    interpreter_context->interpreters->insert(interpreter.get());
+
     EventCounter::IncrementCounter(EventCounter::MessagesConsumed, messages.size());
     CallCustomTransformation(transformation_name, messages, result, accessor, *memory_resource, stream_name);
 
     DiscardValueResultStream stream;
 
     spdlog::trace("Start transaction in stream '{}'", stream_name);
-    utils::OnScopeExit cleanup{[&interpreter, &result]() {
+    utils::OnScopeExit cleanup{[interpreter_context, &interpreter, &result]() {
+      interpreter_context->interpreters->erase(interpreter.get());
```

Comment:
The `CallCustomTransformation` I think can throw an exception. That means this `OnScopeExit` won't be on the stack when we exit the lambda, that means the interpreter won't be deregistered.